pyproject.toml

```diff
@@ -0,0 +1,33 @@
+[tool.poetry]
+name = "sceptre-file-resolver"
+version = "1.0.6"
+description = "A Sceptre resolver to retrieve file content"
+authors = ["zaro0508 <zaro0508@gmail.com>"]
+readme = "README.md"
+homepage = "https://github.com/Sceptre/sceptre-file-resolver"
+license = "Apache-2.0"
+packages = [{ include = "resolver" }]
+keywords = ["sceptre", "sceptre-resolver"]
+classifiers = [
+    "Development Status :: 5 - Production/Stable",
+    "Intended Audience :: Developers",
+    "Natural Language :: English",
+    "Environment :: Console",
+]
+
+[tool.poetry.dependencies]
+python = ">=3.8,<3.12"
```

Comment:
sceptre core supports `3.12` now so maybe change to `">=3.8,<3.13"`?

Reply:
@branchvincent and I have an unanswered question about this bound in [a resolved thread](https://github.com/Sceptre/sceptre-file-resolver/pull/12#discussion_r1417157883).

Does Sceptre need such a tight upper bound?

Why not allow someone to use it with future Python feature versions if they really want to?

The syntax `python = "^3.8"` would allow that.

Reply:
@branchvincent mentioned that the [supported python version classifier is automatically added based on the python requirement](https://python-poetry.org/docs/pyproject/#classifiers)? don't we need the upper bound to restrict the documented support?

Reply:
I think it only generates metadata for the Python versions you have installed. So if you don't install 3.13 in the build environment it won't include that version.

I will show a proper repro when I have a moment.

Reply:
You don't need the upper bound for the classifiers, poetry only generates a classifier for each stable python version meeting your `python` constraint at the time your package is built/published. So for `^3.8`:

```console
$ poetry build -q && tar -xf dist/*tar.gz -O | grep "Programming Language"
Classifier: Programming Language :: Python :: 3
Classifier: Programming Language :: Python :: 3.8
Classifier: Programming Language :: Python :: 3.9
Classifier: Programming Language :: Python :: 3.10
Classifier: Programming Language :: Python :: 3.11
Classifier: Programming Language :: Python :: 3.12
```

Reply:
[The Poetry documentation](https://python-poetry.org/docs/pyproject#classifiers) doesn't give much detail on how this works.

Thanks @branchvincent for the demo. I see the same behavior here.

List available Python feature versions.

```bash
compgen -c | grep -P '^python3\.\d+$' | sort -V -u
```

I still have Python 3.7 installed, and I haven't installed Python 3.12 yet.

```text
python3.7
python3.8
python3.9
python3.10
python3.11
```

Build a Python package from Poetry's new project template.

```bash
cd "$(mktemp --dir)"
poetry new . --quiet
poetry build --quiet
```

Check the Python constraint.

```bash
cat pyproject.toml | grep -P "dependencies|python = "
```

Python 3.8 is the minimum Python version. All later Python 3 versions are allowed.

```toml
[tool.poetry.dependencies]
python = "^3.8"
```

Check the classifiers.

```bash
unzip -q -c dist/*.whl | grep "Programming Language"
```

Because the minimum Python version is 3.8 Poetry excludes the classifier for Python 3.7.

Because the environment doesn't have Python 3.12 Poetry excludes the classifier for Python 3.12.

```text
Classifier: Programming Language :: Python :: 3
Classifier: Programming Language :: Python :: 3.8
Classifier: Programming Language :: Python :: 3.9
Classifier: Programming Language :: Python :: 3.10
Classifier: Programming Language :: Python :: 3.11
```

Reply:
thanks for testing @iainelder.  i guess just `python = "^3.8"` would be fine @branchvincent